src/Component/src/Metadata/Extractor/PhpFileMetadataExtractor.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Resource\Metadata\Extractor;
+
+use Sylius\Resource\Metadata\ResourceMetadata;
+use Symfony\Component\Finder\Finder;
+
+final class PhpFileMetadataExtractor implements MetadataExtractorInterface
+{
+    public function __construct(
+        private readonly array $resourceMapping,
+    ) {
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function extract(): array
+    {
+        $metadata = [];
+
+        foreach ($this->getResourceFilePaths() as $filePath) {
+            if (!is_readable($filePath)) {
+                continue;
+            }
+
+            $resource = $this->getPHPFileClosure($filePath)();
+
+            if (!$resource instanceof ResourceMetadata) {
+                continue;
+            }
+
+            $metadata[] = $resource;
+        }
+
+        return $metadata;
+    }
+
+    private function getResourceFilePaths(): iterable
+    {
+        foreach ($this->createFinder() as $file) {
+            yield $file->getPathname();
+        }
+    }
+
+    private function createFinder(): Finder
+    {
+        $finder = (new Finder())->files();
+
+        foreach ($this->resourceMapping['imports'] ?? [] as $path) {
+            $finder->in($path);
+        }
+
+        return $finder->files();
+    }
+
+    /**
+     * Scope isolated include.
+     *
+     * Prevents access to $this/self from included files.
+     */
+    private function getPHPFileClosure(string $filePath): \Closure
+    {
+        return \Closure::bind(function () use ($filePath): mixed {
+            return require $filePath;
```

Comment:
that's why we cannot import the mapping paths here.
require Classes, Trait, Interface and things like that is not possible. It needs to be basic php file.